sleuth_backend/views/views.py

```diff
@@ -27,70 +29,162 @@ def search(request):
     }
+
     '''
     if request.method != 'GET':
         return HttpResponse(status=405)
 
     core = request.GET.get('core', '')
     query = request.GET.get('q', '')
+    state = request.GET.get('state', '')
+
+    return_fields = 'id,updatedAt,name,description'
+    requested_return_fields = request.GET.get('return', '')
+    if requested_return_fields is not '':
+        return_fields = return_fields + ',' + requested_return_fields
     kwargs = {
         'sort': request.GET.get('sort', ''),
         'start': request.GET.get('start', ''),
         'rows': request.GET.get('rows', ''),
-        'default_field': 'content',
-        'return_fields': 'id,siteName,updatedAt,pageName,pageTitle',
-        'highlight_fields': 'content',
+        'return_fields': return_fields,
     }
 
-    if core is '' or query is '':
+    if query is '':
         sleuth_error = SleuthError(ErrorTypes.INVALID_SEARCH_REQUEST)
         return HttpResponse(sleuth_error.json(), status=400)
 
-    try:
-        query_response = SOLR.query(core, query, **kwargs)
-        response = json.loads(query_response)
-    except pysolr.SolrError as e:
-        sleuth_error = SleuthError(ErrorTypes.SOLR_SEARCH_ERROR, str(e))
-        return HttpResponse(sleuth_error.json(), status=400)
-    except KeyError:
-        sleuth_error = SleuthError(ErrorTypes.UNEXPECTED_SERVER_ERROR)
-        return HttpResponse(sleuth_error.json(), status=500)
-    except ValueError:
-        sleuth_error = SleuthError(ErrorTypes.SOLR_CONNECTION_ERROR)
-        return HttpResponse(sleuth_error.json(), status=500)
-
-    if 'error' in response:
-        sleuth_error = SleuthError(
-            ErrorTypes.SOLR_SEARCH_ERROR,
-            message=response['error']['msg']
-        )
-        return HttpResponse(sleuth_error.json(), status=response['error']['code'])
-    return HttpResponse(query_response)
+    cores_to_search = []
+    if core is '':
+        for c in SOLR.core_names():
+            cores_to_search.append(c)
+    else:
+        cores_to_search.append(core)
+
+    return_fields_list = return_fields.split(",")
+    responses = {
+        'data':[]
+    }
+    for core_to_search in cores_to_search:
+        try:
+            new_query, new_kwargs = _build_search_query(core_to_search, query, kwargs)
+            query_response = SOLR.query(core_to_search, new_query, **new_kwargs)
+            if 'error' in query_response:
+                sleuth_error = SleuthError(
+                    ErrorTypes.SOLR_SEARCH_ERROR,
+                    message=query_response['error']['msg']+" on core "+core_to_search
+                )
+                return HttpResponse(sleuth_error.json(), status=query_response['error']['code'])
+            
+            # Attach type to response and flatten single-item list fields
+            query_response['type'] = core_to_search
+            for doc in query_response['response']['docs']:
+                for f in return_fields_list:
+                    try:
```

Comment:
Rather than using a try-catch here, just check if the field is in `doc` before trying to access it.